strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/repository/NugetRepositoryFeatures.java

```diff
@@ -115,4 +147,70 @@ protected Configuration getConfiguration()
         return configurationManager.getConfiguration();
     }
 
+    @Component
+    public class RepositorySearchEventListener implements CommonEventListener<RemoteRepositorySearchEvent>
```

Comment:
nested @Component may be more difficult to look for, consider top-level class

Reply:
purpose of inner class (and nested component) is to use methods from enclosing class

Reply:
I don't think you need nested class here since you don't call any protected/private method of `NugetRepositoryFeatures` inside `RepositorySearchEventListener`. `RepositorySearchEventListener` may be a top-level class with injected `NugetRepositoryFeatures`  too.

Reply:
yes, it can be a separate class. But I think that this is not a principle. In this case this can well be a nested component.